pycheribuild/projects/cross/llvm_test_suite.py

```diff
@@ -63,11 +63,7 @@ def dependencies(cls, config) -> "tuple[str, ...]":
     @classproperty
     def llvm_project(self) -> "type[BuildLLVMBase]":
         target_info = self.get_crosscompile_target().target_info_cls
-        if issubclass(target_info, FreeBSDTargetInfo):
-            # noinspection PyProtectedMember
-            return target_info._get_compiler_project()
-        else:
-            return BuildCheriLLVM
+        return BuildCheriLLVM
```

Comment:
This looks wrong, should just pass `self.config` to _get_compiler_project()

Reply:
I think this is confusing. The method is a classproperty but it is using `self` as the argument name, which violates PEP8 and should be named `cls`.

Apart from this nitpick, I think we need to deal with the fact that the LLVM project can't be known without a configuration object at this stage.

Reply:
Ok I pushed a fix, it is unclear to me at this point whether we need to patch the compiler_rt and libcxx targets, which seem to use a similar pattern.